test/FactorReTerm.jl

```diff
@@ -99,6 +99,55 @@ end
         form = apply_schema(f, schema(f, slp, contrasts), LMM)
         @test StatsModels.termvars(form.rhs) == [:days, :subj]
     end
+
+    @testset "Runtime construction of random effects terms" begin
+        # operator precedence and basic terms:
+        @test term(:a) | term(:b) isa RandomEffectsTerm
+        @test term(1) + term(:a) | term(:b) isa RandomEffectsTerm
+        @test term(1) + term(:a) + term(:a) & term(:c) | term(:b) isa RandomEffectsTerm
+        
+        # sleep study data:
+        r, d, s, one = term.((:reaction, :days, :subj, 1))
+
+        f1 = @formula(reaction ~ 1 + (1 + days | subj))
+        f2 = r ~ one + (one + d | s)
+        @test f2.rhs[end] isa RandomEffectsTerm
+        ff1 = apply_schema(f1, schema(slp), LMM)
+        ff2 = apply_schema(f2, schema(slp), LMM)
+        # equality of RE terms not defined so check that they generate same modelcols
```

Comment:
should we define equality? I have no idea if it's useful beyond tests....

Reply:
equality of terms generally is kind of a mess so I'm happy to leave it as is for now.